pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java

```diff
@@ -284,7 +284,7 @@
      * @return List of all children of type targetType. Returns an empty list if
      *         none found.
      */
-    <T> List<T> findDescendantsOfType(Class<T> targetType);
+    <T> List<T> findDescendantsOfType(Class<? extends T> targetType);
```

Comment:
@oowekyala I've not merged this into pmd/7.0.x, as this would have created too many follow up changes (e.g. in NodeStream). I think, this change is useful and we should tackle it in a separate change and doing this for the other methods here in Node as well (findChildrenOfType, getParentsOfType, ...). It will then also change the API: https://github.com/pmd/pmd/wiki/PMD-7.0.0-API#description-of-node-api (at least the doc, I assume this change is compatible).